### PR TITLE
Dcerpc udp fixes/v1

### PR DESCRIPTION
**Pull request by inashivb**

### CODE REVIEW

**Comment by victorjulien** [comment]
Please specify the correct SV pr in the body of the pull request.

**Comment by inashivb** [comment]
There is none. These fixes do not seem to be doing any changes to existing output. But, I'm curious to see if they help with any of the perf issues. Peter has been testing this so we'll know. Even if these changes don't fix the issues, these are probably good to fix anyway.

File: rust/src/dcerpc/dcerpc_udp.rs [hunk 1]
```diff
@@ -208,9 +209,24 @@ impl DCERPCUDPState {
 
 #[no_mangle]
 pub extern "C" fn rs_dcerpc_udp_parse(
-    _flow: *mut core::Flow, state: &mut DCERPCUDPState, _pstate: *mut std::os::raw::c_void,
+    _flow: *mut core::Flow, state: *mut std::os::raw::c_void, pstate: *mut std::os::raw::c_void,
     input: *const u8, input_len: u32, _data: *mut std::os::raw::c_void, _flags: u8,
 ) -> AppLayerResult {
+    let eof = unsafe {
+        if AppLayerParserStateIssetFlag(pstate, APP_LAYER_PARSER_EOF) > 0 {
```
Comment (victorjulien) [inline rust/src/dcerpc/dcerpc_udp.rs:216]: Do you have a test that triggers this condition?

I'm a bit worried about about this on C side
```c
/* Flags for AppLayerParserState. */
// flag available                               BIT_U8(0)
#define APP_LAYER_PARSER_NO_INSPECTION          BIT_U8(1)
#define APP_LAYER_PARSER_NO_REASSEMBLY          BIT_U8(2)
#define APP_LAYER_PARSER_NO_INSPECTION_PAYLOAD  BIT_U8(3)
#define APP_LAYER_PARSER_BYPASS_READY           BIT_U8(4)
#define APP_LAYER_PARSER_EOF_TS                 BIT_U8(5)
#define APP_LAYER_PARSER_EOF_TC                 BIT_U8(6)
```
As you can see the EOF has been split into 2 flags, see 4f73943df915832b8aadad853b2c593f7100aa4d

**Comment by inashivb** [comment]
Replaced with #5851

**Pull request closed without merging**